### ISSUE TITLE
Resource name ends up in Rack rather than renaming Rails controller.

### ISSUE THREAD

**Issue by viraptor** [DataDog/dd-trace-rb#2094]
(This may be related to https://github.com/DataDog/dd-trace-rb/issues/2092)

**Current behaviour**

When I try to rename a specific request's span, the name of the `rack` resource changes. With this code:
```
Datadog::Tracing.active_span&.resource = "ApiChunk:#{transaction_name}"
```
Rails lists both `GET 200` and `ApiChunk:user`.

**Expected behaviour**

This actually renames the rails span rather than the rack one, which is what I'd like to do if issue#2092 was fixed. But if I should expect the controller names to be visible in the rack service, then I'd like the naming to happen there.
Basically, I expect to see `ApiChunk:user` in the same service I see `UserController#show`, not in the one listing `GET 200`. 

**Environment**

* **ddtrace version:** 1.1.0
* **Configuration block (`Datadog.configure ...`):**
```
...
  c.tracing.instrument :rails, service_name: 'marketplace'
  c.tracing.instrument :rack, request_queuing: true
...
```
* **Ruby version:** 2.7
* **Operating system:** linux

**Comment by delner** [comment]
I don't think this is a bug. It's likely the Rack, not the Rails span, is active, thus you're annotating its behavior instead. I would check where you're changing this resource name and what the active span is when you do.
